src/actions/chatActions.js

```diff
@@ -267,24 +269,57 @@ export const getChatByContactAction = (
       .then(JSON.parse)
       .catch(() => []);
 
-    const updatedMessages = await receivedMessages.map((message, index) => ({
-      _id: `${message.serverTimestamp}_${index}`,
-      text: message.content,
-      createdAt: new Date(message.serverTimestamp),
-      status: message.status,
-      type: message.type,
-      user: {
-        _id: message.username,
-        name: message.username,
-        avatar,
-      },
-    }))
+    const updatedMessages = receivedMessages
+      .map((message, index) => ({
+        _id: `${message.serverTimestamp}_${index}`,
+        text: message.content,
+        createdAt: new Date(message.serverTimestamp),
+        status: message.status,
+        type: message.type,
+        user: {
+          _id: message.username,
+          name: message.username,
+          avatar,
+        },
+      }))
       .sort((a, b) => b.createdAt - a.createdAt);
 
     dispatch({
       type: UPDATE_MESSAGES,
       payload: { messages: updatedMessages, username },
     });
+
+    if (updatedMessages.length) {
+      const {
+        user: { name: lastMessageSenderUsername },
+        text,
+        createdAt,
+      } = updatedMessages[0];
+
+      if (lastMessageSenderUsername) {
+        dispatch({
+          type: RESET_UNREAD_MESSAGE,
+          payload: {
+            username,
+            lastMessage: {
+              content: text,
+              username: lastMessageSenderUsername,
+              device: 1,
+              serverTimestamp: createdAt,
+              savedTimestamp: 0,
+            },
+          },
+        });
+      }
+    }
+
+    const otherUnreadMessages = existingChats.filter(
```

Comment:
It looks like we don't need all of the messages, but we only want to know if there's any, right? If that's the case, we could use `find` instead.

Reply:
👍 fixed